nix/haskell-packages.nix

```diff
@@ -0,0 +1,126 @@
+{ compiler, enableDhall, enableSwagger }:
+
+pkgsNew: pkgsOld:
+
+let
+  gitignoreSource =
+    let
+      source = pkgsNew.fetchFromGitHub {
+        owner = "hercules-ci";
+        repo = "gitignore.nix";
+        rev = "211907489e9f198594c0eb0ca9256a1949c9d412";
+        sha256 = "06j7wpvj54khw0z10fjyi31kpafkr6hi1k0di13k1xp8kywvfyx8";
+      };
+    in (import source { inherit (pkgsNew) lib; }).gitignoreSource;
+
+in {
+  haskellPackages = pkgsOld.haskell.packages."${compiler}".override (old: {
+    overrides =
+      pkgsOld.lib.composeExtensions
+        (old.overrides or (_: _: {}))
+        (haskellPackagesNew: haskellPackagesOld:
+          let
+            proto3-suite-flags = (if enableDhall then ["-fdhall"] else []) ++ (if enableSwagger then [] else ["-f-swagger"]);
+
+          in {
+            tasty = haskellPackagesNew.callPackage ./haskell-packages/tasty.nix { };
+
+            tasty-ant-xml = haskellPackagesNew.callPackage ./haskell-packages/tasty-ant-xml.nix { };
```

Comment:
I would really like to avoid checking in `cabal2nix` codegen. Please use [`packageSourceOverrides`](https://github.com/NixOS/nixpkgs/blob/7e9b0dff974c89e070da1ad85713ff3c20b0ca97/pkgs/development/haskell-modules/lib.nix#L43-L53) in a new extension, or call [`callCabal2nix` or `callHackage`](https://github.com/NixOS/nixpkgs/blob/7e9b0dff974c89e070da1ad85713ff3c20b0ca97/pkgs/development/haskell-modules/make-package-set.nix#L182-L220) directly, instead of `callPackage` on generated Nix code.

Reply:
I would add this to the `haskellPackages` extension:

```nix
"range-set-list" = pkgsOld.haskell.lib.overrideCabal haskellPackagesOld."range-set-list" (old: {
  broken = false;
  jailbreak = true;
});
```